plugin-api/src/main/java/org/hyperledger/besu/plugin/services/storage/KeyValueStorage.java

```diff
@@ -61,15 +62,23 @@
   Optional<byte[]> get(byte[] key) throws StorageException;
 
   /**
-   * Performs an evaluation against each key in the store, keeping the entries that pass, removing
-   * those that fail.
+   * Returns a stream of all keys.
    *
-   * @param retainCondition predicate to evaluate each key against, unless the result is {@code
-   *     null}, both the key and associated value must be removed.
-   * @return the number of keys removed.
-   * @throws StorageException problem encountered when removing data.
+   * @return A stream of all keys in storage.
+   * @throws StorageException problem encountered during the retrieval attempt.
+   */
+  Stream<byte[]> streamKeys() throws StorageException;
+
+  /**
+   * Delete the value corresponding to the given key if a write lock can be instantly acquired on
+   * the underlying storage. Do nothing otherwise.
+   *
+   * @param key The key to delete.
+   * @throws StorageException any problem encountered during the deletion attempt.
+   * @return false if the lock on the underlying storage could not be instantly acquired, true
+   *     otherwise
    */
-  long removeAllKeysUnless(Predicate<byte[]> retainCondition) throws StorageException;
```

Comment:
Sanity check - I assume we've thought through the implications of removing this public API?

Reply:
We added it just for pruning and I think the only dependency for the storage plugin api is PegaSys plus. I'll ask them if they still want it around.

Reply:
Since this could be a breaking change if anyone depends on this API, might be worth moving this API clean-up into a separate PR, so we can be more careful about when / how exactly we release this.